### PR TITLE
Improve importers logic

### PR DESCRIPTION
**Pull request by mwestphal**
- vtkImporter::Update now returns bool, use it
 - vtkImporter::UpdateAtTimeValue has been added, use it
 - Add a new vtkF3DImporter for easier subclassing in plugins

### CODE REVIEW

File: library/testing/CMakeLists.txt [hunk 1]
```diff
@@ -3,8 +3,8 @@ list(APPEND libf3dSDKTests_list
      TestSDKEngine.cxx
      TestSDKEngineExceptions.cxx
      TestSDKOptions.cxx
-     TestSDKLog.cxx
      TestSDKLoader.cxx
+     TestSDKLog.cxx
```
Comment (mwestphal) [inline library/testing/CMakeLists.txt:7]: revert order

File: testing/data/duck_invalid.gltf
Comment (mwestphal) [inline testing/data/duck_invalid.gltf:1]: add info in DATA_LICENSES.md

File: vtkext/private/module/vtkF3DGenericImporter.cxx [hunk 2]
```diff
@@ -394,16 +400,35 @@ std::string vtkF3DGenericImporter::GetMetaDataDescription()
...
+    {
+      F3DLog::Print(F3DLog::Severity::Warning, "A reader failed to update at a timeValue");
+      pipe.Output = nullptr;
+      pipe.Imported = false;
+      continue;
+    }
+    hasGeometry = true;
+  }
+
+  if (hasGeometry)
```
Comment (mwestphal) [inline vtkext/private/module/vtkF3DGenericImporter.cxx:423]: incorrect, should be done everytime instead

File: plugins/assimp/module/vtkF3DAssimpImporter.cxx [hunk 3]
```diff
@@ -602,7 +602,8 @@ class vtkF3DAssimpImporter::vtkInternals
     }
     catch (const DeadlyImportError& e)
     {
-      vtkWarningWithObjectMacro(this->Parent, "Assimp exception: " << e.what());
+      vtkErrorWithObjectMacro(this->Parent, "Assimp exception: " << e.what());
+      return false;
```
Comment (Meakk) [inline plugins/assimp/module/vtkF3DAssimpImporter.cxx:606]: Cannot be covered?

**Review by Meakk — COMMENTED** [review]
Ok that seems reasonable.  
I think it would be good to have a superclass for all F3D importers (generic/usd/assimp) in the public vtk module to handle all the VTK version compatibility stuff to avoid duplication and help writing a new importer. What do you think?

**Comment by Meakk** [comment]
Also it looks that wasm is broken (most likely because of the VTK update)

File: plugins/assimp/module/vtkF3DAssimpImporter.cxx (hunk 3 above)
Comment (mwestphal) [reply plugins/assimp/module/vtkF3DAssimpImporter.cxx:606]: According to Assimp, no, but I was not able to trigger that code anyway.

**Comment by mwestphal** [comment]
> I think it would be good to have a superclass for all F3D importers (generic/usd/assimp) in the public vtk module to handle all the VTK version compatibility stuff to avoid duplication and help writing a new importer. What do you think?

This is reasonnable and I actually considered it initially. Ill do it.

File: plugins/assimp/module/vtkF3DAssimpImporter.cxx (hunk 3 above)
Comment (mwestphal) [reply plugins/assimp/module/vtkF3DAssimpImporter.cxx:606]: I tried modifying the fbx binary manually but always hit other errors, never this one, even when trying to read assimp code.

**Review by Meakk — APPROVED** [review]

**Pull request merged**